spec/codecs/identity_map_codec_spec.rb

```diff
@@ -271,7 +271,7 @@
           l_2 = Mlc::LineListener.new(queue, imc, "stream3").tap {|o| o.accept("baz")}
         end
         it "three events are auto flushed from three different identities/codecs" do
-          sleep 0.6 # wait for auto_flush - in multiples of 0.5 plus 0.1
+          sleep 1 # wait for auto_flush 2 PeriodicRunner cycles to ensure it is executed before assertion.
           expect(queue.size).to eq(3)
```

Comment:
May we can make it a bit more fast to run, but still provide resilience by doing retries with exponential backoff:

```suggestion
          sleep 0.6 # wait for auto_flush 1 PeriodicRunner cycles before assertion.
          # try a few times with exponential backoff as timing is not 100% guaranteed
          try(10) { expect(queue.size).to eq(3) }
```


Result:

```
❯ bundle exec rspec -e 'when wrapped codec has auto-flush activated' -fd
[...]

Randomized with seed 50649
LogStash::Codecs::IdentityMapCodec
  observer/listener based processing
    normal processing
      when wrapped codec has auto-flush activated
Failed (
expected: 3
     got: 0

(compared using ==)
). Sleeping for 0.02
Failed (
expected: 3
     got: 0

(compared using ==)
). Sleeping for 0.04
Failed (
expected: 3
     got: 0

(compared using ==)
). Sleeping for 0.08
Failed (
expected: 3
     got: 0

(compared using ==)
). Sleeping for 0.16
        three events are auto flushed from three different identities/codecs

Finished in 0.65658 seconds (files took 1.08 seconds to load)
1 example, 0 failures

Randomized with seed 175
```

Reply:
Great idea!

Reply:
Suggestion applied and CI 🟢 again.